xinference/model/rerank/core.py

```diff
@@ -268,6 +268,8 @@ def rerank(
             similarity_scores = self._model.compute_score(sentence_combinations)
             if not isinstance(similarity_scores, Sequence):
                 similarity_scores = [similarity_scores]
+            if similarity_scores and isinstance(similarity_scores[0], Sequence):
```

Comment:
This looks like just the opposite operation of 269-270.

if similarity_scores is not a sequence, wrap into a list, then check if similarity_scores[0] is a sequence, unwrap it.

Reply:
Yes, there are different strange structures in the return values ​​of bge series models

Reply:
So I just wonder if there is more clear implementation?

Reply:
This is the simplest, most painless fix I can think of, maybe you guys can have a more elegant solution to fix this problem

Reply:
```
if not isinstance(similarity_scores, Sequence):
    similarity_scores = [similarity_scores]
elif isinstance(similarity_scores, list) and len(similarity_scores) > 0  and isinstance(similarity_scores[0], Sequence):
```

I think `elif` is more clear because they cannot happen at the same time. About the condition I modified a bit, you can see if it can work.

Reply:
Optimized the implementation of the code based on your suggestions

Reply:
Thanks, have you verified it can work?